docs/source/ops/monitoring.rst

```diff
@@ -0,0 +1,130 @@
+Monitoring
+==========
+
+This section gives a high-level overview of the PrivateStorageIO monitoring efforts.
+
+
+Goals
+`````
+
+Alerting
+  Something might break soon, so somebody needs to do something.
+
+Comparing over time or experiment groups
+  Is our service slower than it was last week? Does database B answer queries faster than database A?
+
+Analyzing long-term trends
+  How big is my database and how fast is it growing? How quickly is my daily-active user count growing?
+
+
+Introduction to our dashboards
+``````````````````````````````
+
+We have two groups of dashboards: Requests (external view, RED method) and Resources (internal view, USE method).
+
+Services and their dependencies can be visualized as a tree from external-facing to internal systems.
+
+.. figure:: DAG-of-services.svg
+    :width: 200px
+    :align: center
+    :alt: DAG of services, created using https://www.nomnoml.com/
+    :figclass: align-center
+
+    DAG of services
+
+We order our dashboards like a breadth-first-search of that tree.
+This makes it easier to understand dependencies and faster to trouble shoot when a high-latency problem on a low-level service bubbles up.
+
+
+Meaning of our metrics
+``````````````````````
+
+Google's *Monitoring Distributed Systems* book about what they call the `Four Golden Signals <https://sre.google/sre-book/monitoring-distributed-systems/#xref_monitoring_golden-signals>`_ has a great explanation and definition of useful metrics:
+
+Latency
+  Requests, but also errors take time, so don't discard them.
+
+Traffic
+  What constitutes "Traffic" depends on the nature of your system.
+
+Errors
+  (The *rate* of ) failed requests.
+
+Saturation
+  How "full" your service is.  Take action (i.e. page a human) before service degrades.
+
+"If you measure all four golden signals and page a human when one signal is problematic (or, in the case of saturation, nearly problematic), your service will be at least decently covered by monitoring."
+

```

Comment:
My reading of this section is that we will use these four signals to guide our choices about metrics to monitor, information to present on dashboards, and alerts to emit.

As we do start to add specific metrics I think it might be cool if we called back to this section to explain the decision to include them.

(nothing actionable here)